src/utils/writeToCsv.spec.ts

```diff
@@ -0,0 +1,140 @@
+          Object.values(csvColumns)
+        )
+      }
+
+      const csvContent = await fileModule.readFile(csvFilePath)
+
+      const row1 =
+        ',firstColumn-1,secondColumn-1,thirdColumn-1,fourthColumn-1,fifthColumn-1'
+      const row2 =
+        ',firstColumn-2,secondColumn-2,thirdColumn-2,fourthColumn-2,fifthColumn-2'
+      const row3 =
+        ',firstColumn-3,secondColumn-3,thirdColumn-3,fourthColumn-3,fifthColumn-3'
+
+      expect(csvContent).toEqual(expect.stringContaining(row1))
+      expect(csvContent).toEqual(expect.stringContaining(row2))
+      expect(csvContent).toEqual(expect.stringContaining(row3))
+    })
+  })
+
+  describe('should throw error', () => {
+    afterEach(() => {
+      jest.clearAllMocks()
+    })
+
+    it(`should throw 'Error while checking if csv file exists.'`, async () => {
+      jest.spyOn(fileModule, 'fileExists').mockRejectedValueOnce('')
+      try {
+        await writeToCsv(csvFilePath, data, Object.values(csvColumns))
+        expect(false).toBe(true)
+      } catch (e) {
+        expect(e).toEqual('Error while checking if csv file exists.\n')
+      }
+    })
+
+    it(`should throw 'Error while creating CSV file.' if file does not exists already`, async () => {
+      jest
+        .spyOn(fileModule, 'fileExists')
+        .mockImplementationOnce(() => Promise.resolve(false))
+      jest.spyOn(fileModule, 'createFile').mockRejectedValueOnce('')
+
+      try {
+        await writeToCsv(csvFilePath, data, Object.values(csvColumns))
+        expect(false).toBe(true)
+      } catch (e) {
+        expect(e).toEqual('Error while creating CSV file.\n')
+      }
+    })
+
+    it(`should throw 'Error while reading CSV file.'`, async () => {
+      jest.spyOn(fileModule, 'readFile').mockRejectedValueOnce('')
+
+      try {
+        await writeToCsv(csvFilePath, data, Object.values(csvColumns))
+        expect(false).toBe(true)
+      } catch (e) {
+        expect(e).toEqual('Error while reading CSV file.\n')
+      }
+    })
+
+    it(`should throw 'Error while creating CSV file.' when creating file finally`, async () => {
+      jest
+        .spyOn(fileModule, 'createFile')
+        .mockImplementationOnce(async (fileName: string, content: string) => {
+          fileName = fileModule.unifyFilePath(fileName)
+          if (fileName.split(path.sep).length > 1) {
+            let folderPathParts = fileName.split(path.sep)
+            folderPathParts.pop()
+            const folderPath = folderPathParts.join(path.sep)
+
+            if (!(await fileModule.folderExists(folderPath))) {
+              await fileModule.createFolder(folderPath)
+            }
+          }
+
+          return fs.promises.writeFile(fileName, content)
+        })
+        .mockRejectedValueOnce('')
+
+      try {
+        await writeToCsv(csvFilePath, data, Object.values(csvColumns))
+        expect(false).toBe(true)
+      } catch (e) {
+        expect(e).toEqual('Error while creating CSV file.\n')
+      }
```

Comment:
Need to remove try catch and use following syntax
`await expect().rejects.toEqual(...)`